src/main.rs

```diff
@@ -107,78 +122,100 @@ fn mem_parser(arg: &str) -> Result<usize, String> {
   Ok(base * mult)
 }
 
-impl WOpts {
-  fn get_warning_opts(self) -> WarningOpts {
-    let mut warning_opts = WarningOpts::default();
-    let argm = Args::command().get_matches();
-    if let Some(wopts_id_seq) = argm.get_many::<clap::Id>("WOpts") {
-      let allows = &mut self.allows.into_iter();
-      let denies = &mut self.denies.into_iter();
-      let warns = &mut self.warns.into_iter();
-      WarningOpts::from_cli_opts(&mut warning_opts, wopts_id_seq.collect(), allows, denies, warns);
-    }
-    warning_opts
-  }
-}
-
 fn main() {
   fn run() -> Result<(), String> {
     #[cfg(not(feature = "cli"))]
     compile_error!("The 'cli' feature is needed for the hvm-lang cli");
 
-    let args = Args::parse();
-    let mut opts = Opts::light();
-    Opts::from_vec(&mut opts, args.opts)?;
-    let warning_opts = args.wopts.get_warning_opts();
-
-    let mut book = load_file_to_book(&args.path)?;
-    if args.verbose {
-      println!("{book:?}");
-    }
+    let cli = Cli::parse();
+    let opts = Opts::light();
```

Comment:
It could instead be called inside `from_cli_opts`, like this:
```
fn from_cli_opts(cli_opts) -> Self {
  let mut opts = Opts::light();
  for cli_opt in cli_opts {
     match cli_opt {...}
  }
}
```